### PR TITLE
Rule evaluator business telemetry

### PR DESCRIPTION
**Pull request by jhrozek**
Extends the telemetry store so that all the rules being evaluated are stored in the telemetry store for sending later.

Fixes: #1881

### CODE REVIEW

**Comment by jhrozek** [comment]
Still a draft until we merge #2101 but rebased atop that PR, with some tests etc..

**Comment by JAORMX** [comment]
@jhrozek if you're gonna move the entity info wrapper ( which I think is a good idea) I suggest doing so in a dedicated PR

**Comment by jhrozek** [comment]
> @jhrozek if you're gonna move the entity info wrapper ( which I think is a good idea) I suggest doing so in a dedicated PR

Done. I think this PR is now ready for review.

File: internal/engine/executor.go [hunk 1]
```diff
@@ -380,6 +399,7 @@ func (e *Executor) releaseLockAndFlush(
 func logEval(
 	ctx context.Context,
 	inf *entities.EntityInfoWrapper,
+	profileName string,
```
Comment (rdimitrov) [inline internal/engine/executor.go:402]: This is already part of the evalParams, i.e. `params.Profile.Name`

File: internal/engine/executor.go [hunk 2]
```diff
@@ -407,6 +427,10 @@ func logEval(
 		Str("action", "alert").
 		Str("action_status", string(evalerrors.ErrorAsAlertStatus(params.GetActionsErr().AlertErr))).
 		Msg("result - action")
+
+	// log business logic
+	minderlogger.BusinessRecord(ctx).
+		AddRuleEval(params.Rule.Type, profileName, params)
```
Comment (rdimitrov) [inline internal/engine/executor.go:433]: It's not going to be that explicit, but I think you already have the rule and profile names through params (the ActionsParams interface)

**Review by JAORMX — COMMENTED** [review]
telemetry store parts lgtm.

**Comment by jhrozek** [comment]
> telemetry store parts lgtm.

Thanks for the review, I'll address @rdimitrov's concerns as well

File: internal/engine/executor.go (hunk 1 above)
Comment (jhrozek) [reply internal/engine/executor.go]: addressed in the new patch

File: internal/engine/executor.go (hunk 2 above)
Comment (jhrozek) [reply internal/engine/executor.go]: thanks, addressed in the new patch

File: internal/engine/interfaces/interface.go [hunk 3]
```diff
@@ -76,6 +76,10 @@ const (
 	ActionOptUnknown
 )
 
+func (a ActionOpt) String() string {
+	return [...]string{"on", "off", "dry_run", "unknown"}[a]
```
Comment (JAORMX) [inline internal/engine/interfaces/interface.go:80]: what is this witchery

**Review by JAORMX — APPROVED** [review]

File: internal/engine/interfaces/interface.go (hunk 3 above)
Comment (jhrozek) [reply internal/engine/interfaces/interface.go:80]: it just says that the size of the array is determined by the number of the elements

**Pull request merged**